lib/sqlalchemy/dialects/mysql/reflection.py

```diff
@@ -416,7 +416,7 @@ def _prep_regexes(self):
             r"(?: +KEY_BLOCK_SIZE *[ =]? *(?P<keyblock>\S+))?"
             r"(?: +WITH PARSER +(?P<parser>\S+))?"
             r"(?: +COMMENT +(?P<comment>(\x27\x27|\x27([^\x27])*?\x27)+))?"
-            r"(?: +/\*(?P<version_sql>.+)\*/ +)?"
+            r"(?: +/\*(?P<version_sql>.+)\*/ *)?"
```

Comment:
this is not the correct token in which to reflect the "PRIMARY KEY".   I believe the term "PRIMARY" has to be matched by the `(?P<type>\S+)` group.      otherwise, all we are doing is suppressing a warning for something that will be incorrectly ignored in any case.

Reply:
```PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */```
results in
```{'type': 'PRIMARY', 'name': None, 'using_pre': None, 'columns': '`id`', 'using_post': None, 'keyblock': None, 'parser': None, 'comment': None, 'version_sql': 'T![clustered_index] CLUSTERED '}```
The `type` seems correct to me and so does `columns` etc.

`comment` is None, which is also correct as there is no `COMMENT=` etc.

The `version_sql` looks slightly weird as this is where the version specific MySQL comments are later filtered out (e.g. `/*!5723 WITH PARSER(foo) */` but I don't think that's a problem as `self._re_key_version_sql` is handling this correctly by not matching on this.

Reply:
if you can add a specific test for this asserting that the match tokens go where they are expected, that would be appreciated.

Reply:
I added this to the test:
```diff
+        m = regex.match("  PRIMARY KEY (`id`)")
+        eq_(m.group("type"), "PRIMARY")
+        eq_(m.group("columns"), "`id`")
```